### PR TITLE
docker-1 set basic configuration

### PR DESCRIPTION
**Pull request by Davvidos**

### CODE REVIEW

File: nginx/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+server {
+    listen 80 default_server;
+    root /home/restapi/http/public;
+
+    index index.php index.html index.htm;
```
Comment (jgrygierek) [inline nginx/nginx.conf]: I'm not sure if adding more than one index name has any sense. Just put here your real index name.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    volumes:
+      - ../rest-api:/home/restapi/http
+    networks:
+      - app-network
+
+  db:
+    container_name: mysql
+    image: mysql:5.7
+    ports:
+      - "3306:3306"
```
Comment (jgrygierek) [inline docker-compose.yml]: You can have more than 1 docker running, so in this case I prefer configurable ports. Just use here env variable. `${DOCKER_MYSQL_PORT}:3306`

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+version: '3'
+services:
+  nginx:
+    image: nginx:latest
+    container_name: nginx
+    ports:
+      - "80:80"
```
Comment (jgrygierek) [inline docker-compose.yml]: The same, you can use configurable ports.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+version: '3'
+services:
+  nginx:
```
Comment (jgrygierek) [inline docker-compose.yml:3]: If you will have more than 1 docker with the same names of services, you can have warnings about orphaned services. You can add `COMPOSE_PROJECT_NAME=your_app_name` to your env variables used by docker, so all your services will have names starting with this app name.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    networks:
+      - app-network
+
+  db:
+    container_name: mysql
+    image: mysql:5.7
+    ports:
+      - "3306:3306"
+    environment:
+      MYSQL_ROOT_PASSWORD: root
```
Comment (jgrygierek) [inline docker-compose.yml]: do not hardcode password, use .env file.

File: docker-compose.yml (hunk 4 above)
Comment (jgrygierek) [reply docker-compose.yml:3]: .

File: .env [hunk 6]
```diff
@@ -0,0 +1,7 @@
+DOCKER_NGINX_PORT=80:80
```
Comment (jgrygierek) [inline .env]: Just add here your external port (left part). Right part can be hardcoded in docker compose

File: .env [hunk 7]
```diff
@@ -0,0 +1,7 @@
+DOCKER_NGINX_PORT=80:80
+DOCKER_MYSQL_PORT=3306:3306
```
Comment (jgrygierek) [inline .env]: Just add here your external port (left part). Right part can be hardcoded in docker compose

**Review by jgrygierek — CHANGES_REQUESTED** [review]

File: .env (hunk 6 above)
Comment (Davvidos) [reply .env]: @jgrygierek done, please check it now.

File: .env (hunk 7 above)
Comment (Davvidos) [reply .env]: @jgrygierek done, please check it now.

**Review by jgrygierek — APPROVED** [review]

**Pull request merged**